configure.ac

```diff
@@ -526,6 +526,89 @@ if test "x$enable_unwind" = xno; then
    AC_SEARCH_LIBS([backtrace], [execinfo])
 fi
 
+AC_ARG_ENABLE([backtrace],
+              [AS_HELP_STRING([--enable-backtrace=<unwind-ptrace>],
+              [enable printing backtraces.])],
+              [],
+              [enable_backtrace=no])
+if test "x$enable_backtrace" = xyes; then
+   case $my_htop_platform in
+      linux)
+         enable_backtrace=unwind-ptrace
+         ;;
+      *)
+         AC_MSG_ERROR([backtrace feature not implemented for the '$my_htop_platform' platform])
+         ;;
+   esac
+fi
+
+case "$enable_backtrace" in
+   no)
+      ;;
+   unwind-ptrace)
+      AC_CHECK_LIB([unwind-ptrace], [_UPT_create], [], [AC_MSG_ERROR([can not find required library libunwind-ptrace and libunwind-generic])], [-lunwind-generic])
+      AC_CHECK_LIB([unwind-generic], [_UPT_create], [], [AC_MSG_ERROR([can not find required library libunwind-generic])])
+      AC_CHECK_HEADERS([libunwind-ptrace.h], [], [
+         CPPFLAGS="$CPPFLAGS -I/usr/include/libunwind"
+         AC_CHECK_HEADERS([libunwind-ptrace.h], [], [AC_MSG_ERROR([can not find required header file libunwind-ptrace.h])])
+      ])
+      ;;
+   *)
+      AC_MSG_ERROR([bad value '$enable_backtrace' for --enable-backtrace])
+      ;;
+esac
+if test "x$enable_backtrace" != xno; then
+   AC_DEFINE([HAVE_BACKTRACE], [1], [Define if the backtrace feature is enabled])
+fi
+AM_CONDITIONAL([HAVE_BACKTRACE], [test x"$enable_backtrace" != xno])
+
+if test "x$enable_backtrace" = xunwind-ptrace; then
+   AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
+   #include <libunwind-ptrace.h>
+   #ifndef unw_get_elf_filename
+   # error 'unw_get_elf_filename' not defined
+   #endif
+   ]])], [AC_DEFINE([HAVE_LIBUNWIND_ELF_FILENAME], [1], [Define if libunwind has unw_get_elf_filename])], [])
```

Comment:
Isn't there some existing AutoConf magic to do this?

Reply:
I cannot use `AC_CHECK_LIB` because `unw_get_elf_filename` is a macro. I searched and it's the only method I found.

Reply:
@MrCirdo Use `AC_LINK_IFELSE` then, and add a comment that `unw_get_elf_filename` may be a macro.

Reply:
It's done. Thank you @Explorer09

Reply:
No no no. The reason for `AC_LINK_IFELSE` is that it checks the linking with the library as well as compiling. So you should write a simple program that calls the function (be it a macro or not). You can use `LDFLAGS` and `LIBS` variables along with `AC_LINK_IFELSE`.

Reply:
Oh okay, I see. If I understand correctly. I wrote that : 
```
if test "x$enable_backtrace" = xunwind-ptrace; then
   # Check if the macro unw_get_elf_filename exists
   AC_LINK_IFELSE(
     [AC_LANG_PROGRAM(
       [[#include <libunwind-ptrace.h>]],
       [[unw_get_elf_filename(0, 0, 0, 0);]]
     )],
     [AC_DEFINE([HAVE_LIBUNWIND_ELF_FILENAME], [1], [Define if libunwind has unw_get_elf_filename])],
     []
   )
fi
```

Reply:
@MrCirdo You partly get the idea. This checks if the macro exists and whether it can link to libunwind successfully with the said function.

Allow me to polish the code a bit (I add a `AC_MSG_CHECKING` so that when the compiling fails, it's easier to look up the error in the config log file; also I add type casts which might be helpful):
```m4
if test "x$enable_backtrace" = xunwind-ptrace; then
   AC_MSG_CHECKING([if libunwind has unw_get_elf_filename])
   AC_LINK_IFELSE(
      [AC_LANG_PROGRAM(
         [[/* unw_get_elf_filename might exist as a macro */
         #include <libunwind-ptrace.h>]],
         [[unw_get_elf_filename((unw_cursor_t*)0, (char*)0, (size_t)0, (unw_word_t*)0);]]
      )],
      [AC_DEFINE([HAVE_LIBUNWIND_ELF_FILENAME], [1], [Define if libunwind has unw_get_elf_filename])
      AC_MSG_RESULT(yes)],
      [AC_MSG_RESULT(no)]
   )
fi
```

Reply:
Thank you very much @Explorer09, I added your code 😄 
I pushed some changes about the demangling (strongly inspired by your code). Is it good for you?